lib/util/registerExternalSerializer.js

```diff
@@ -279,9 +279,14 @@ register(
 		 * @returns {void}
 		 */
 		serialize(source, { write }) {
-			const data = source.sourceAndMap({});
-			write(data.source);
-			write(JSON.stringify(data.map));
+			if (source._innerSourceMap) {
```

Comment:
Could you please give some background about this change? What does it match? What’s the reasoning here?

Reply:
Sure. I've created a small demo repo [here](https://github.com/SevInf/sourcemap-source-demo/blob/master/demo.js).

If you run this file you'll see that without inner source map, `sourceMapSource.sourceAndMap()` produces a result, identical to the inputs (the only exception is `file` field of source map which webpack ignores to my knowledge). `sourceAndMap` call however, boils down to [`SourceNode. fromStringWithSourceMap`](https://github.com/mozilla/source-map/blob/master/lib/source-node.js#L55) call, which is quite expensive. Skipping it allows to save a lot of serialization times (~2 min on our codebase).